R/LoadDataPage.R

```diff
@@ -51,6 +50,37 @@ load_data_page_server <- function(id, metaoutcome) {
     long_format_upload_panel_server(id = 'long_upload')
     wide_format_upload_panel_server(id = 'wide_upload')
     
-    return(list(data = data, treatment_list = treatment_list))
+    find_all <- function(data, field_name) {
+      if (field_name %in% colnames(data)) {
+        # Long format
+        return(unique(data[[field_name]]))
+      } else {
+        # Wide format
+        all <- c()
+        for (i in seq(6)) {
+          all <- c(all, data[[paste0(field_name, ".", i)]])
+        }
+        return(unique(all))
+      }
+    }
+    
+    treatment_list <- reactive({
+      treatment_names = find_all(data(), "T")
+      return(data.frame(Number = seq(1, length(treatment_names)), Label = treatment_names))
+    })
+    
+    wrangled_data <- reactive({
+      df <- isolate(data())
```

Comment:
`isolate` prevents the reactive from taking a dependency on the `data()` reactive. This is done to simplify the reactivity graph. Since the `treatment_list()` reactive is already dependent on `data()`, I decided that it makes more sense that the reactivity should follow the chain:
- Data uploaded
- Treatment IDs generated
- Data wrangled to use the generated IDs

The assignment of `treatment_ids` is not neccessary, but I think that it makes coneptual sense that the reactive funciton is only called once, making it clear that the value is accessed once, then used, rather than being multiple calls to a function which could, in theory, change value between calls.